src/overlays/actors/ovl_En_Sda/z_en_sda.c

```diff
@@ -28,18 +32,327 @@ const ActorInit En_Sda_InitVars = {
+                        temp_v1 = temp_t0 + phi_a0;
+                        if ((temp_v1 >= 0) && (temp_v1 < 0x40)) {
+                            temp_v0 = temp_t1 + phi_a3;
+                            if ((temp_v0 >= 0) && (temp_v0 < 0x1000)) {
+                                shadowTexture[temp_v1 + temp_v0] = 255;
+                            }
+                        }
+                    }
+                }
+            } else if (D_80947AEC[i] == 0) {
+                for (j = 0, phi_a3 = -0xC0; j < 7; j++, phi_a3 += 0x40) {
+                    for (phi_a0 = -D_80947A78[j]; phi_a0 < D_80947A78[j] - 1; phi_a0++) {
+                        temp_v1 = temp_t0 + phi_a0;
+                        if ((temp_v1 >= 0) && (temp_v1 < 0x40)) {
+                            temp_v0 = temp_t1 + phi_a3;
+                            if ((temp_v0 >= 0) && (temp_v0 < 0x1000)) {
+                                shadowTexture[temp_v1 + temp_v0] = 255;
+                            }
+                        }
+                    }
+                }
+            } else if (D_80947AEC[i] == 4) {
+                for (j = 0, phi_a3 = -0x1C0; j < 14; j++, phi_a3 += 0x40) {
+                    for (phi_a0 = -D_80947AB0[j]; phi_a0 < D_80947AB0[j]; phi_a0++) {
+                        temp_v1 = temp_t0 + phi_a0;
+                        if ((temp_v1 >= 0) && (temp_v1 < 0x40)) {
+                            temp_v0 = temp_t1 + phi_a3;
+                            if ((temp_v0 >= 0) && (temp_v0 < 0x1000)) {
+                                shadowTexture[temp_v1 + temp_v0] = 255;
+                            }
+                        }
+                    }
+                }
+            } else {
+                for (j = 0, phi_a3 = -0x80; j < 6; j++, phi_a3 += 0x40) {
+                    for (phi_a0 = -D_80947A6C[j]; phi_a0 < D_80947A6C[j] - 1; phi_a0++) {
+                        temp_v1 = temp_t0 + phi_a0;
+                        if ((temp_v1 >= 0) && (temp_v1 < 0x40)) {
+                            temp_v0 = temp_t1 + phi_a3;
+                            if ((temp_v0 >= 0) && (temp_v0 < 0x1000)) {
+                                shadowTexture[temp_v1 + temp_v0] = 255;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+void func_8094702C(EnSda* this, u8* shadowTexture, Player* player, GlobalContext* globalCtx) {
+    s16 temp_t0;
+    s16 temp_t1;
+    s16 temp_v0;
+    s16 temp_v1;
+    s16 phi_a0;
+    s16 phi_a3;
+    s16 i;
+    s16 j;
+    Vec3f sp194;
+    Vec3f sp188;
+    s32* shadowTextureTemp32;
+    u8* shadowTextureTemp;
+    Vec3s sp178;
+    Vec3f sp16C;
+    Vec3f sp64[22];
+
+    if (BREG(57) != 0) {
+        for (shadowTextureTemp = shadowTexture, i = 0; i < 0x1000; i++, shadowTextureTemp++) {
+            if ((i >= 0 && i < 0x40) || (i >= 0xFC0 && i < 0x1000) || ((i & 0x3F) == 0) || ((i & 0x3F) == 0x3F)) {
```

Comment:
Some hardcoded shadow values... why not just start with a texture, they had plenty of rom space.